irobotclient/client.py

```diff
@@ -0,0 +1,75 @@
+"""
+Copyright (c) 2017 Genome Research Ltd.
+
+This program is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation, either version 3 of the License, or (at your
+option) any later version.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import requests
+
+from irobotclient.interface import Interface
+
+SUCCESS_RESPONSE = 200
+WAIT_RESPONSE = 202, 206
+MATCHED_RESPONSE = 304
+
+
+class IrobotClient(Interface):
+
+    def __init__(self, command_line_args):
+        Interface.__init__(self, command_line_args)
+
```

Comment:
I use vim for python and use this : https://github.com/python-mode/python-mode. It automatically does pep8 checking and formatting and it runs a static analyser (pylint) to flag up any obvious syntax errors.